### DISCUSSION TITLE
Briggs & Stratton 26 kW with genmon - Amazing Software!!

### DISCUSSION THREAD

**Discussion by royf007** [jgyates/genmon#1112] [Show and tell]
I recently discovered genmon, and I must say, it's an absolute marvel of software engineering. My gratitude to Mr. Yates for crafting such a remarkable tool! As a fellow software developer, I can only fathom the countless hours and immense effort poured into its creation.

In regard to monitoring solutions, my experience with the Briggs & Stratton offerings, particularly their Amplify InfoHub WiFi module, left much to be desired. It provided scant information, lacked support for VLANs, and tethered me to the same WiFi network as the generator. Additionally, Briggs & Stratton offers no API, nor integrations, such as IFTTT or Home Assistant. In contrast, the built-in genmon web server is nothing short of fantastic. While I nearly opted for the B&G InfoHub cellular monitoring, it too lacks the integrations I sought.

Given my use of Home Assistant for home automation, integrating the MQTT client in genmon was a natural step, and data transmission has been flawless.

The documentation, videos, and support surrounding genmon, along with its robust feature set, are truly outstanding. Setting up genmon with my Briggs & Stratton system was a breeze.

With a strong outdoor WiFi signal, I opted for the USR-W630 and equipped it with a waterproof WiFi antenna. Indoors, I have genmon running on a Raspberry Pi 4B.

For those interested, here's the WiFi antenna I used: [[link]](https://a.co/d/h2ko8T0)  . I still need to drill a 1/2 inch hole on the side of the generator for mounting. Additionally, an SMA adapter is necessary to connect this antenna to the USR-W630: [[link]](https://a.co/d/7IYflHC) .

😁

![genmon_briggs_stratton](https://github.com/jgyates/genmon/assets/4976402/f84d85a7-25c9-4897-8ee7-b956c8607edc)
![IMG_4421](https://github.com/jgyates/genmon/assets/4976402/bcdd5c39-6e77-40a6-9a9b-13fa1da7ded8)
![IMG_4425](https://github.com/jgyates/genmon/assets/4976402/b24ff6a9-10f9-4a1c-9fa3-652d768c8d52)
![IMG_4431](https://github.com/jgyates/genmon/assets/4976402/2f025527-f1f7-445c-b0e5-04ad4ec8c467)

**Comment by jgyates** [comment]
@royf007 

thanks for the info. The links will come in hand for others as well. Also thanks for the feedback and the donation. I am glad it is working well for you. It is always good to hear success stories (even better with pictures).

thanks again,
@jgyates

**Comment by Pharp202** [comment]
@royf007 
Thank you for sharing, I just recently stumbled upon genmon and more specifically for my 22kw Briggs and Stratton. I was curious what type of wire did you use to connect the power and the USR-W630. I have stumbled upon some YouTube videos by someone named genmon Briggs, but he does not go over some of the wiring. Thanks for sharing about the Raspberrypi 4 working.

**Comment by royf007** [comment]
@Pharp202 , sure, for connecting the power and the USR-W630, I used the following low voltage wire: [18 Gauge 3 Conductor](https://a.co/d/9hkmSr8).  It keeps things neater.

The whole setup has been working flawlessly for months now.
